Judge/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/service/JudgeQueueListener.java

```diff
@@ -1,42 +1,29 @@
 package pl.jakubowskiprzemyslaw.tajgertim.service;
 
 import org.springframework.amqp.rabbit.annotation.RabbitListener;
-import org.springframework.stereotype.Service;
-import pl.jakubowskiprzemyslaw.tajgertim.models.round.NextRoundStatus;
-import pl.jakubowskiprzemyslaw.tajgertim.models.round.RoundStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import pl.jakubowskiprzemyslaw.tajgertim.event.PlayerShootResultEvent;
 import pl.jakubowskiprzemyslaw.tajgertim.models.shoot.PlayerShootResult;
-import pl.jakubowskiprzemyslaw.tajgertim.models.shoot.ShootResult;
-import pl.jakubowskiprzemyslaw.tajgertim.queues.Queues;
 import pl.jakubowskiprzemyslaw.tajgertim.services.BattleShipQueueInteractionHandler;
 import pl.jakubowskiprzemyslaw.tajgertim.services.LoggerService;
-import pl.jakubowskiprzemyslaw.tajgertim.services.QueueService;
 
 @BattleShipQueueInteractionHandler
 public class JudgeQueueListener {
 
-    private final QueueService queueService;
     private final LoggerService logger;
+    private final ApplicationEventPublisher publisher;
 
-    public JudgeQueueListener(QueueService queueService, LoggerService logger) {
-        this.queueService = queueService;
+    @Autowired
+    public JudgeQueueListener(LoggerService logger, ApplicationEventPublisher publisher) {
         this.logger = logger;
+        this.publisher = publisher;
     }
 
     @RabbitListener(queues = "JudgePlayerShootResultQueue")
-    public void listenOnJudgePlayerShootResultQueue (PlayerShootResult playerShootResult) {
+    public void listenOnJudgePlayerShootResultQueue(PlayerShootResult playerShootResult) {
         logger.logInfo(JudgeQueueListener.class, "Received message" + playerShootResult);
-
-        ShootResult shootResult = playerShootResult.getShootResult();
-
-        RoundStatus roundStatus = RoundStatus.GAME_END;
-
-        if (shootResult.equals(ShootResult.HIT)) {
-            roundStatus = RoundStatus.SAME_PLAYER;
-        }
-        else if (shootResult.equals(ShootResult.MISS)) {
-            roundStatus = RoundStatus.NEXT_PLAYER;
-        }
-
-        queueService.sendObjectToQueue(Queues._14PlayingStateMachineNextRoundStatusQueue, new NextRoundStatus(roundStatus));
+        PlayerShootResultEvent event = new PlayerShootResultEvent(this, playerShootResult);
```

Comment:
I think it would look better with using factory method rather than plain constructor. Just a style comment.

Reply:
will consider it later